### PR TITLE
Add framework specific environment variables for plugin discovery

### LINKED ISSUE #8151: Introduce `NUGET_NETFX_PLUGIN_PATHS` and `NUGET_NETCORE_PLUGIN_PATHS` to support configuration of both at same time
NUGET_PLUGIN_PATHS can't be configured statically, to work for both nuget.exe and dotnet.  If you add paths for the .exe and .dll credential providers to handle both cases, nuget fails trying to execute a dll and dotnet fails trying to execute the full framework .exe.

## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe):
**nuget.exe, dotnet.exe**
NuGet version (x.x.x.xxx):
**5.0.0.5856**
dotnet.exe --version (if appropriate):
**2.2.300**
VS version (if appropriate):

OS version (i.e. win10 v1607 (14393.321)):
**win10 v1903**
Worked before? If so, with which NuGet version:
**no**
## Detailed repro steps so we can see the same problem

1. `set nuget_plugin_paths=d:\tools\credprovider\plugins\netcore\CredentialProvider.Microsoft\CredentialProvider.Microsoft.dll;d:\tools\credprovider\plugins\netfx\CredentialProvider.Microsoft\CredentialProvider.Microsoft.exe`
2. (fails) `nuget.exe install newtonsoft.json`
3. (fails) `dotnet new web & dotnet add package newtonsoft.json`

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/8151  
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Replaces   https://github.com/NuGet/NuGet.Client/pull/2859/files. 

NUGET_PLUGIN_PATHS can't be configured statically, to work for both nuget.exe and dotnet. If you add paths for the .exe and .dll credential providers to handle both cases, nuget fails trying to execute a dll and dotnet fails trying to execute the full framework .exe.

A few approaches were considered: 

* Extend on the convention based discovery, as described in previous PR and issue. 
Basically prefer .dll for netcore and .exe for netfx. 
The concern with this approach is dotnet core 3.0 SDK creates exes by default for console netcoreapp3.0 projects. Another concern is eliminating the ability to disable all plugins by specifying " ". 

* Add 2 new framework specific environment variables. Those take precedence over the generic variable.

The 2nd is the approach we are implementing. 

Docs changes: https://github.com/NuGet/docs.microsoft.com-nuget/pull/1599

//cc @zjrunner @zarenner 

## Testing/Validation

Tests Added: Yes
Reason for not adding tests:  
Validation:

### CODE REVIEW

**Comment by zarenner** [comment]
Looks great, excited to have this. Thanks Nikolche.

**Review by zarenner — APPROVED** [review]

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs [hunk 1]
```diff
@@ -304,8 +304,15 @@ private async Task<Lazy<IPluginMulticlientUtilities>> PerformOneTimePluginInitia
             {
                 throw new ArgumentNullException(nameof(pluginFactoryCreator));
             }
-
-            _rawPluginPaths = reader.GetEnvironmentVariable(EnvironmentVariableConstants.PluginPaths);
+#if IS_DESKTOP
+            _rawPluginPaths = reader.GetEnvironmentVariable(EnvironmentVariableConstants.DesktopPluginPaths);
+#else
+            _rawPluginPaths = reader.GetEnvironmentVariable(EnvironmentVariableConstants.CorePluginPaths);
+#endif
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs:311]: If we make our core libraries, like `NuGet.Protocol` only target `netstandard2.0` in the future, and let the .NET Framework use the .NET Standard versions, then this will stop working. We need a runtime check to see if we're on the .NET Framework on .NET Core.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs:311]: NuGet.Protocol depends on Packaging, so it will keep cross targeting. 
But yeah we'll need to account for conditional compilation in the projects that will switch to only 1 framework.

**Review by zivkan — APPROVED** [review]
I still think it would be good to change the `#if` to a runtime check now, rather than later when we try to reduce our TFMs.

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs:311]: I'd argue separation of concerns :) 
It's a problem for when we try to reduce our frameworks. 
Note that I'm 100% confident we won't need to do that with Plugins and NuGet.Protocol is higher on the dependency graph where .NET Framework APIs are just required due to Credentials needing them.
